src/Schema/Comparator.php

```diff
@@ -21,23 +21,29 @@
 class Comparator
 {
     /**
+     * Returns a SchemaDiff object containing the differences between the schemas $fromSchema and $toSchema.
+     *
+     * The returned differences are returned in such a way that they contain the
+     * operations to change the schema stored in $fromSchema to the schema that is
+     * stored in $toSchema.
+     *
+     * This method is marked as static to avoid a BC break but should be called
+     * non-statically since it will not be static in the next major release.
```

Comment:
For this, we can try adding a runtime deprecation based on `if (! isset($this))`.

Reply:
I tried the `if (! isset($this))` trick: it only works when the method is not marked as static.

But `ComparatorTest` fails when the `static` keyword is removed:
> Non-static method Doctrine\DBAL\Schema\Comparator::compareSchemas() should not be called statically

And it looks a lot like a BC break to me...

Reply:
As a workaround, I guess we could make the method private and leverage `__call()` and `__callStatic` to add the runtime deprecation, but this is not ideal.

What do you think?

Reply:
`__call` and `__callStatic` is the only way possible to make both ways work AFAIK.

Reply:
Could this method be named differently to prevent this issue and possible confusion when refactoring to the new API?

Reply:
Another option would be to name the new method `compareSchemata` as “schemata” is also a plural form of “schema”

Reply:
A runtime deprecation isn't a must and it looks like it cannot be implemented here. We definitely cannot remove the `static` flag, so let's leave it as is and just document the static call deprecation.

> `__call` and `__callStatic` is the only way possible to make both ways work AFAIK.

I wouldn't complicate it that much.

> Another option would be to name the new method `compareSchemata` as “schemata” is also a plural form of “schema”

While this is right, it would be inconsistent with the other method names. I'd prefer to keep the original name.